src/hooks/useUndernames/useUndernames.tsx

```diff
@@ -0,0 +1,260 @@
+import { useEffect, useState } from 'react';
+
+import { useArweaveCompositeProvider, useIsMobile } from '..';
+import {
+  CalendarTimeIcon,
+  ChevronUpIcon,
+  NotebookIcon,
+  PencilIcon,
+  TargetIcon,
+  TrashIcon,
+} from '../../components/icons/index';
+import CopyTextButton from '../../components/inputs/buttons/CopyTextButton/CopyTextButton';
+import {
+  ArweaveTransactionID,
+  PDNTContractJSON,
+  UndernameMetadata,
+} from '../../types';
+import eventEmitter from '../../utils/events';
+
+export function useUndernames(id: ArweaveTransactionID) {
+  const isMobile = useIsMobile();
+  const arweaveDataProvider = useArweaveCompositeProvider();
+  const [sortAscending, setSortOrder] = useState(true);
+  const [sortField, setSortField] = useState<keyof UndernameMetadata>('status');
+  const [selectedRow, setSelectedRow] = useState<UndernameMetadata>(); // eslint-disable-line
+  const [rows, setRows] = useState<UndernameMetadata[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [percent, setPercentLoaded] = useState<number | undefined>();
+
+  useEffect(() => {
+    if (id) {
```

Comment:
As `id` is not nullable, and is the instance of a class, this conditional will always return true. Consider either removing the conditional or making the parameter to be nullable